tg_bot/modules/bans.py

```diff
@@ -191,6 +191,66 @@ def unban(bot: Bot, update: Update, args: List[str]) -> str:
     return log
 
 
+@run_async
+@bot_admin
+def rban(bot: Bot, update: Update, args: List[str]):
+    message = update.effective_message
+
+    if args:
+        user_id, chat_id = extract_user_and_text(message, args)
+
+        if not user_id:
+            message.reply_text("You don't seem to be referring to a user.")
+            return ""
+        elif not chat_id:
+            message.reply_text("You don't seem to be referring to a chat.")
+            return ""
+
+        try:
+            chat = bot.get_chat(chat_id)
+        except BadRequest as excp:
+            if excp.message == "Chat not found":
+                message.reply_text("Chat not found! Make sure you entered a valid chat ID")
+                return ""
+            else:
+                raise
+
+    else:
+        message.reply_text("You don't seem to be referring to a chat/user.")
+        return ""
+
+    try:
+        member = chat.get_member(user_id)
+    except BadRequest as excp:
+        if excp.message == "User not found":
+            message.reply_text("I can't seem to find this user")
+            return ""
+        else:
+            raise
+
+    if is_user_ban_protected(chat, user_id, member):
+        message.reply_text("I really wish I could ban admins...")
+        return ""
+
+    if user_id == bot.id:
+        message.reply_text("I'm not gonna BAN myself, are you crazy?")
+        return ""
+
+    try:
+        chat.kick_member(user_id)
+        message.reply_text("Banned!")
+    except BadRequest as excp:
+        if excp.message == "Reply message not found":
```

Comment:
You're going to need more exception catching here, see the possible gban exceptions.